lib/resource/Directory.js

```diff
@@ -75,7 +77,19 @@ Directory.prototype.getProvider = function getProvider(/* [options,] callback */
     return callback();
   }
 
-  return this.dataStore.getResource(this.provider.href, options, require('./Provider'), callback);
+  function providerFactory(provider) {
```

Comment:
...but that doesn't really solve it either. It's not possible to send in a factory with the current structure.